operator/cmd/mesh/manifest-apply.go

```diff
@@ -206,3 +220,37 @@ func ApplyManifests(setOverlay []string, inFilenames []string, force bool, dryRu
 
 	return saveIOPToCluster(reconciler, iopStr)
 }
+
+// DetectIstioVersionDiff will show warning if istioctl version and control plane version are different
+// nolint: interfacer
+func DetectIstioVersionDiff(cmd *cobra.Command, obv string, maArgs *manifestApplyArgs) error {
+	var opts clioptions.ControlPlaneOptions
+	icps, err := GetVersionInfo(maArgs.kubeConfigPath, maArgs.context, opts)
+	if err != nil {
+		return err
+	}
+	if len(*icps) != 0 {
+		var icpVer string
+		for _, icp := range *icps {
+			// Get the version without a hash for comparison
+			// 1.6.4-58f551e08f08addfd81783e1c2ed1eabeb836168
+			versionParts := strings.Split(icp.Info.Version, "-")
+			if obv != versionParts[0] {
+				icpVer = versionParts[0]
+			}
+		}
+		if obv != icpVer {
+			cmd.Printf("! istioctl version %s and the control plane version %s are different.\n", obv, icpVer)
+		}
+	}
+	return nil
+}
+
+// GetVersionInfo returns control plan version
+func GetVersionInfo(kubeconfig string, configContext string, opts clioptions.ControlPlaneOptions) (*istioVersion.MeshInfo, error) {
+	kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), opts.Revision)
```

Comment:
Why are we creating a new client just for this? We should have 1 kube client at the top of the command and use it throughout

Reply:
Moved `kubeclient` to the top level. But I think `ApplyManifests`  need different set of params and requires more refactoring. I want to get this merged and I can do refactoring in separate PR?